connectd/peer_exchange_initmsg.c

```diff
@@ -110,10 +114,24 @@ static struct io_plan *peer_init_received(struct io_conn *conn,
 	if (tlvs->remote_addr) {
 		/* 'parsing' can be done by overlaying the correct C struct */
 		remote_addr_any = (struct remote_addr_any *)tlvs->remote_addr;
-		status_peer_debug(&peer->id,
-				  "Peer reported remote_addr type %i. "
-				  "TODO: do something useful",
-				  remote_addr_any->type);
+		if (remote_addr_any->type == ADDR_TYPE_IPV4) {
+			remote_addr_v4 = (struct remote_addr_ipv4 *)remote_addr_any;
+			remote_addr_wi->itype = ADDR_INTERNAL_WIREADDR;
+			remote_addr_wi->u.wireaddr.type = ADDR_TYPE_IPV4;
+			remote_addr_wi->u.wireaddr.port = remote_addr_v4->port;
+			remote_addr_wi->u.wireaddr.addrlen = 4;
+			memcpy(&remote_addr_wi->u.wireaddr.addr,
+			       &remote_addr_v4->addr, 4);
+		}
```

Comment:
If you did proper marshal/demarshal you wouldn't have a buffer overrun here in the case where the buffer is malformed.

"remote_addr" is a weird name in general, it's "address peer sees me as", or something...

Reply:
Yep will do the demarshalling functions. Regarding the name: I called it this way because the RFC also uses this name: https://github.com/lightning/bolts/pull/917  which makes sense from the protocol viewpoint, but not so much from the receiving peer.